sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -26,6 +26,25 @@ import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.execution.SparkPlan
 import org.apache.spark.sql.types.StructType
 
+private class BatchIterator[T](iter: Iterator[T], batchSize: Int)
+  extends Iterator[Iterator[T]] {
+
+  override def hasNext: Boolean = iter.hasNext
+
+  override def next(): Iterator[T] = {
```

Comment:
The behavior of this `BatchIterator` seems to be a bit weird and seems to not follow `Iterator`. For example, you can continue call its `next()` and never consume it to the end.

Reply:
If we really don't want to use `GroupedIterator` as suggested in https://github.com/apache/spark/pull/18732/files#r142577976, we should add a clear warning comment on it.

Reply:
+1. Let me add that.

Reply:
Added.

Reply:
I didn't see the comment added?

Reply:
Sorry I pushed a bit late. The comment is added now.